tests/contrib/hooks/test_snowflake_hook.py

```diff
@@ -71,3 +108,30 @@ def test_get_conn_params(self):
 
     def test_get_conn(self):
         self.assertEqual(self.db_hook.get_conn(), self.conn)
+
+    def test_key_pair_auth_encrypted(self):
+        self.conn.extra_dejson = {'database': 'db',
+                                  'account': 'airflow',
+                                  'warehouse': 'af_wh',
+                                  'region': 'af_region',
+                                  'role': 'af_role',
+                                  'private_key_file': '/tmp/test_key.p8'}
```

Comment:
You have already prepared `self.encryptedPrivateKey = "/tmp/test_key.p8"` above, so I believe it's better to refer `self.encryptedPrivateKey` rather than hardcoding here.

Reply:
Fixed.